node-src/lib/findChangedDependencies.ts

```diff
@@ -60,29 +60,44 @@ export const findChangedDependencies = async (ctx: Context) => {
 
   ctx.log.debug({ pathPairs }, `Found ${pathPairs.length} manifest/lockfile pairs to check`);
 
-  const tracedPairs = pathPairs.filter(([manifestPath, lockfilePath]) => {
-    if (untraced.some((glob) => matchesFile(glob, manifestPath))) return false;
-    if (untraced.some((glob) => matchesFile(glob, lockfilePath))) return false;
-    return true;
-  });
-  const untracedCount = pathPairs.length - tracedPairs.length;
-  if (untracedCount) {
-    ctx.log.debug(`Skipping ${untracedCount} manifest/lockfile pairs due to --untraced`);
+  // Now filter out any pairs that don't have git changes, or for which the manifest is untraced
+  const filteredPathPairs = pathPairs
+    .map(([manifestPath, lockfilePath]) => {
+      const commits = packageMetadataChanges
+        .filter(
+          ({ changedFiles }) =>
+            changedFiles.includes(manifestPath) || changedFiles.includes(lockfilePath)
+        )
+        .map(({ commit }) => commit);
```

Comment:
We should probably dedupe `commits`.

Reply:
I believe the original list is one per commit?

Reply:
Wouldn't it be possible to have multiple `packageMetadataChanges` for the same build (i.e. commit)? E.g. when there's two package.json files in the repo and they both changed.

Reply:
No, because the changes are the list of *all* the metadata changes for that commit. So it'd be an entry `{ changedFiles: ['package.json', 'subdir/package.json'], commit: 'xyz' }`. 

See: https://github.com/chromaui/chromatic-cli/blob/5436e2db7c0a1227799c5b4cf2f7bb7c843751ba/node-src/tasks/gitInfo.ts#L197-L208

Although... maaaaybe it is possible to have >1 build for the same commit?